server/src/main/java/org/opensearch/index/store/RemoteDirectory.java

```diff
@@ -61,6 +66,32 @@ public Collection<String> listFilesByPrefix(String filenamePrefix) throws IOExce
         return blobContainer.listBlobsByPrefix(filenamePrefix).keySet();
     }
 
+    public List<String> listFilesByPrefixInLexicographicOrder(String filenamePrefix, int limit) throws IOException {
+        List<String> sortedBlobList = new ArrayList<>();
+        AtomicReference<Exception> exception = new AtomicReference<>();
+        blobContainer.listBlobsByPrefixInSortedOrder(
+            filenamePrefix,
+            limit,
+            BlobContainer.BlobNameSortOrder.LEXICOGRAPHIC,
+            new ActionListener<>() {
+                @Override
+                public void onResponse(List<BlobMetadata> blobMetadata) {
+                    sortedBlobList.addAll(blobMetadata.stream().map(BlobMetadata::name).collect(Collectors.toList()));
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    exception.set(e);
+                }
+            }
+        );
+        if (exception.get() == null) {
+            return sortedBlobList;
```

Comment:
we are taking assumption on `blobContainer.listBlobsByPrefixInSortedOrder` being sync with a listener, based on the current implementation. Is that a valid assumption to make ?

Reply:
Let me change the implementation.

Reply:
Changed to `LatchedActionListener`